examples/name-service/programs/name-service-without-macros/src/lib.rs

```diff
@@ -0,0 +1,231 @@
+use std::net::{Ipv4Addr, Ipv6Addr};
+
+use anchor_lang::prelude::*;
+use borsh::{BorshDeserialize, BorshSerialize};
+use light_hasher::bytes::AsByteVec;
+use light_sdk::account_info::LightAccountInfo;
+use light_sdk::{
+    compressed_account::{LightAccount, LightAccounts},
+    instruction_data::LightInstructionData,
+    light_system_accounts,
+    verify::verify_compressed_accounts,
+    LightDiscriminator, LightHasher, LightTraits,
+};
+
+declare_id!("7yucc7fL3JGbyMwg4neUaenNSdySS39hbAk89Ao3t1Hz");
+
+#[program]
+pub mod name_service {
+    use light_hasher::Discriminator;
+    use light_sdk::account_info::convert_metas_to_infos;
+
+    use super::*;
+
+    pub fn create_record<'info>(
+        ctx: Context<'_, '_, '_, 'info, CreateRecord<'info>>,
+        inputs: Vec<u8>,
+        name: String,
+        rdata: RData,
+    ) -> Result<()> {
+        let inputs = LightInstructionData::deserialize(&inputs)?;
+        let mut account_infos = convert_metas_to_infos(&inputs.accounts, &crate::ID)?;
+
+        account_infos[0].derive_address(
+            &[b"name-service", name.as_bytes()],
+            &crate::ID,
+            ctx.remaining_accounts,
+        )?;
+        account_infos[0].set_discriminator(NameRecord::discriminator());
+
+        let mut light_accounts = LightCreateRecord::try_light_accounts(&account_infos)?;
+
+        light_accounts.record.owner = ctx.accounts.signer.key();
+        light_accounts.record.name = name;
+        light_accounts.record.rdata = rdata;
+
+        verify_compressed_accounts(
+            &ctx,
+            inputs.proof,
+            &[light_accounts.record],
+            None,
+            false,
+            None,
+            &crate::ID,
+        )?;
+
+        Ok(())
+    }
+
+    pub fn update_record<'info>(
+        ctx: Context<'_, '_, '_, 'info, UpdateRecord<'info>>,
+        inputs: Vec<u8>,
+        new_rdata: RData,
+    ) -> Result<()> {
+        let inputs = LightInstructionData::deserialize(&inputs)?;
+        let mut account_infos = convert_metas_to_infos(&inputs.accounts, &crate::ID)?;
+
+        account_infos[0].set_discriminator(NameRecord::discriminator());
+
+        let mut light_accounts = LightCreateRecord::try_light_accounts(&account_infos)?;
```

Comment:
why don't we deserialize `try_light_accounts` from `account_meta` directly?
(I think this way this will work more naturally with the discriminators by putting them into the `try_light_accounts` implementation.)

Reply:
`LightAccount<T>` has a `LightAccountInfo` as a field and honestly, I would prefer it to stay that way - thanks to that, `LightAccount` can reuse methods of `LightAccountInfo` - like `input_compressed_accounts` and `output_compressed_accounts`.

`LightAccount<T>` in fact just adds the Borsh bits on top of `LightAccoutInfo` and overwrites the output data with what user did in the `account_state` field. The goal is to do only that and nothing more there.